ray-operator/controllers/ray/common/pod.go

```diff
@@ -365,6 +368,15 @@ func BuildPod(ctx context.Context, podTemplateSpec corev1.PodTemplateSpec, rayNo
 		cmd += convertCmdToString(pod.Spec.Containers[utils.RayContainerIndex].Args)
 	}
 
+	if rayNodeType == rayv1.HeadNode && gcsOptions != nil {
+		if gcsOptions.RedisUsername != nil {
+			rayStartParams["redis-username"] = "$REDIS_USERNAME"
+		}
+		if gcsOptions.RedisPassword != nil {
+			rayStartParams["redis-password"] = "$REDIS_PASSWORD"
+		}
+	}
```

Comment:
If the new gcsOptions is specified with redis username and password, we add --redis-username and --redis-password flags to the ray start command.

The $REDIS_USERNAME and $REDIS_PASSWORD will be replaced with env vars by the shell.